skill-tree /src/test/java/com/RDS/skilltree/SkillsIntegrationTests.java

```diff
@@ -1,137 +1,250 @@
+                .body(skillDRO)
+                .post("/v1/skills/");
 
-        /*
+        response.then()
+                .statusCode(400)
+                .body("error", equalTo("Bad Request")); //Todo change this on introducing the global exception handling
+    }
+
+    @Test
+    @DisplayName("Return 400, if name is not passed for Skill creation")
+    public void testAPIReturns400_OnNameNotPassedForSkillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .type(SkillType.ATOMIC)
+                .createdBy(user.getId())
+                .build();
 
-        [
-          {
-            "skillName": "Java",
-            "userName": "John doe"
-          }
-        ]
-         */
+        Response response = given()
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
 
         response.then()
-                .statusCode(200)
+                .statusCode(400)
+                .body("error", equalTo("Bad Request"));
+    }
+
+    @Test
+    @DisplayName("Return 409, if name is already used for Skill creation")
+    public void testAPIReturns409_OnNameAlreadyUsedForSkillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .type(SkillType.ATOMIC)
+                .name("Java")
+                .createdBy(user.getId())
+                .build();
+
+        Response response = given()
                 .contentType("application/json")
-                .body("code",equalTo(200))
-                .body("data", hasSize(1))
-                .body("data[0].skillName",equalTo("Java"))
-                .body("data[0].userName",equalTo("John doe"));
+                .body(skillDRO)
+                .post("/v1/skills/");
+
+        response.then()
+                .statusCode(409)
+                .body("message", equalTo("Cannot create entry for Skill as Skill name is duplicate"));
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return 400 on invalid skill name passed")
-    public void testAPIReturns400_OnIncorrectSkillName() {
+    @DisplayName("Return 201, on successful Skill creation")
+    public void testAPIReturns201_OnSuccessfulSkillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .type(SkillType.ATOMIC)
+                .name("Go")
+                .createdBy(user.getId())
+                .build();
+
         Response response = given()
-                .get("/skills/{skillName}");
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
 
         response.then()
-                .statusCode(400)
-                .body("code",equalTo(400))
-                .body("message",equalTo("Invalid skillName passed"));
+                .statusCode(201)
+                .contentType("application/json")
+                .body("name", equalTo("Go"))
+                .body("type", equalTo("ATOMIC"));
```

Comment:
can we also assert that response object has id of the newly created skill

Reply:
I don't think so we can check that, as its a random UUID and we are not passing that value, guide me how can we do it.